Sources/ObservableStore/ObservableStore.swift

```diff
@@ -54,6 +54,45 @@ extension ModelProtocol {
     }
 }
 
+extension ModelProtocol {
+    /// Update a child state within a parent state.
+    /// This update offers a convenient way to call child update functions
+    /// from the parent domain, and get parent-domain states and actions
+    /// back from it.
+    ///
+    /// - `get` gets the child's state
+    /// - `set` sets the child's state within the parent state
+    /// - `tag` tags child actions, turning them into parent actions
+    /// - `state` the outer state
+    /// - `action` the inner action
+    /// - `environment` the environment for the update function
+    /// - Returns a new outer state
+    public static func update<ViewModel: ModelProtocol>(
```

Comment:
This `update(get:set:tag:state:action:environment:)` replaces `CursorProtocol.update`.

We already synthesize `update(state:actions:environment:)` for `ModelProtocol`, so it makes more sense to find the child-updating update function on `ModelProtocol` as well.

Reply:
Passing `get:set:tag` closures offers a more flexible and direct way to map from parent domain to child domain vs the `CursorProtocol` approach. You can still write a cursor struct if you like, and pass its static methods to these arguments.

Dynamic `get:set:tag` also means we can do away with a separate `KeyedCursorProtocol`. Keyed children can be handled via closure.